src/pfs/src/pfsc.rs

```diff
@@ -100,13 +100,13 @@ impl<F: Read + Seek> Reader<F> {
             buf.fill(0);
         } else {
             // Read compressed.
-            let mut compressed = new_buffer(size as usize - 2);
+            let mut compressed = new_buffer(size as usize);
 
-            self.file.seek(SeekFrom::Start(offset + 2))?;
+            self.file.seek(SeekFrom::Start(offset))?;
             self.file.read_exact(&mut compressed)?;
 
             // Decompress.
-            let mut deflate = flate2::Decompress::new(false);
+            let mut deflate = flate2::Decompress::new(true);
```

Comment:
You need to change this to https://docs.rs/flate2/1.0.24/flate2/read/struct.ZlibDecoder.html. See [here](https://github.com/ultimaweapon/obliteration/blob/main/src/pup/src/entry.rs#L233) for how we this with PUP.